src/Resources/contao/widgets/PickerWidget.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao;
+
+
+/**
+ * Provide methods to handle input field "picker".
+ *
+ * @property boolean        $multiple
+ * @property array          $rootNodes
+ * @property string         $fieldType
+ * @property string         $foreignTable
+ * @property string         $context
+ * @property string         $labelField
+ * @property string         $icon
+ * @property boolean        $sortable
+ * @property array|callable $label_callback
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+class PickerWidget extends \Widget
+{
+
+	/**
+	 * Submit user input
+	 * @var boolean
+	 */
+	protected $blnSubmitInput = true;
+
+	/**
+	 * Template
+	 * @var string
+	 */
+	protected $strTemplate = 'be_widget';
+
+
+	/**
+	 * Load the database object
+	 *
+	 * @param array $arrAttributes
+	 */
+	public function __construct($arrAttributes=null)
+	{
+		$this->import('Database');
```

Comment:
Do not import it here, use `getInstance()` instead.

Reply:
I have asked @aschempp  about the style of this *legacy* class with and he said I should just take the existing code (based on page tree widget) as is for now. Ideally we should not use `Database` but the db connection service.

Reply:
Yeah it's fine for me to just use the legacy class here, but still `Database::getInstance()` instead of importing it in some global variable :)